src/lib/Microsoft.Health.Fhir.Ingest.Common/Template/FhirTemplate.cs

```diff
@@ -1,14 +1,18 @@
-﻿// -------------------------------------------------------------------------------------------------
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
-// -------------------------------------------------------------------------------------------------
-
-namespace Microsoft.Health.Fhir.Ingest.Template
-{
-    public abstract class FhirTemplate : IFhirTemplate
-    {
-        public virtual string TypeName { get; set; }
-
-        public virtual ObservationPeriodInterval PeriodInterval { get; set; }
-    }
-}
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Ingest.Template
+{
+    public abstract class FhirTemplate : IFhirTemplate
+    {
+        [JsonProperty(Required = Required.Always)]
+        public virtual string TypeName { get; set; }
```

Comment:
Type is the only required field since this is where determine the binding the class representing the template.

Reply:
Will remove. Thanks for double checking on these requirements.